src/solvers/dgsem_unstructured/sort_boundary_conditions.jl

```diff
@@ -41,10 +41,29 @@ function initialize!(boundary_types_container::UnstructuredSortedBoundaryTypes{N
 
   unique_names = unique(cache.boundaries.name)
 
-  # TODO: This needs to be handled differently for the `ParallelP4estMesh` since the boundaries
-  # are distributed and thus unique_names only contains the names of boundaries on the local process
-  # See https://github.com/trixi-framework/Trixi.jl/issues/1047
-  if !mpi_isparallel()
+  if mpi_isparallel()
+    # Exchange of boundaries names
+    unique_names_byte=Vector{UInt8}(join(unique_names, "\0"))
```

Comment:
I am not a fan of calling variables by the type they are supposed to hold. I'd rather name it for its purpose, e.g.
```suggestion
    send_buffer = Vector{UInt8}(join(unique_names, "\0"))
```